lesson16/src/js/main.js

```diff
@@ -0,0 +1,205 @@
+import { parseISO, differenceInDays } from 'date-fns';
+
+const categoryTab = document.getElementById('js-tab');
+
+function showLoadingImg() {
+	const tabContentsArea = document.getElementById('js-newsContents');
+	const createLoadingBox = document.createElement('div');
+	createLoadingBox.classList.add('loading');
+	createLoadingBox.id = 'js-loading';
+	const loadingImg = document.createElement('img');
+	loadingImg.src = 'images/loading-circle.gif';
+	loadingImg.width = 100;
+	loadingImg.height = 100;
+	loadingImg.classList.add('el_loadingImg');
+	tabContentsArea.appendChild(createLoadingBox).appendChild(loadingImg);
+}
+
+function hideLoadingImg() {
+	document.getElementById('js-loading').remove();
+}
+
+function createTabContentsArea(){
+	const tabContentsArea = document.createElement('div');
+	tabContentsArea.id = 'js-newsContents';
+	categoryTab.parentNode.insertBefore(tabContentsArea,categoryTab.nextSibling);
+}
+
+function createTabContent(data){
+	const tabContentsArea = document.getElementById('js-newsContents');
+	data.forEach((category,i) => {
+		const newsContentArea = document.createElement('div');
+		newsContentArea.classList.add('news-area');
+		newsContentArea.setAttribute('role','tabpanel');
+		newsContentArea.setAttribute('id','panel-'+ (i+1));
+		newsContentArea.setAttribute('aria-labelledby','tab-'+ (i+1));
+		newsContentArea.setAttribute('aria-hidden','true');
+		const articleData = category.article;
+		tabContentsArea.appendChild(newsContentArea).appendChild(renderNewsArticle(articleData));
+	})
+}
+
+function renderCategoryTab(data){
+	const fragment = document.createDocumentFragment();
+
+	data.forEach((target,i) => {
+		const categoryTab = document.createElement('li');
+		categoryTab.classList.add('tab_list_item');
+		categoryTab.setAttribute('role','presentation');
+
+		const categoryButton = document.createElement('button');
+		categoryButton.textContent = target.category;
+		categoryButton.setAttribute('id','tab-' + (i+1));
+		categoryButton.setAttribute('class', 'tab-button js-tabButton');
+		categoryButton.setAttribute('aria-controls','panel-' + (i+1));
+		categoryButton.setAttribute('role', 'tab');
+		categoryButton.setAttribute('aria-selected', 'false');
+		categoryButton.setAttribute('tabindex', '-1');
+		fragment.appendChild(categoryTab).appendChild(categoryButton);
+	});
+	const categoryTabLists = document.querySelector('.tab-list');
+	categoryTabLists.appendChild(fragment);
+}
+
+const renderNewsArticle = (data) => {
```

Comment:
IMO
I thought the function name and the content were different, what do you think? I don't think render is done because it is a function that returns `newsLists`.
I wonder if it's something like this...
`getNewsArticle `

Reply:
Thank you for the review!
Certainly it is not displayed by the function here!
Since it is created based on the data, I changed it to getNewsArticle as advised.

[9aa360b](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/9aa360b66711976b19346c3ef615693094aaa6cb)